airflow/www/static/js/connection_form.js

```diff
@@ -123,6 +125,16 @@ function applyFieldBehaviours(connection) {
  */
 function handleTestConnection(connectionType, testableConnections) {
   const testButton = document.getElementById("test-connection");
+
+  if (!configTestConnectionEnabled) {
+    // If test connection is not enabled in config, disable button and display toolip
+    // alerting the user.
+    $(testButton)
+      .prop("disabled", true)
+      .attr("title", "Test connection is not enabled in config.");
```

Comment:
Another suggestion
```suggestion
      .attr("title", "Testing connections is disabled in Airflow configuration.");
```

Reply:
I wonder if we should just get rid of the button altogether. I think that'd be my vote.

Reply:
I think It's kinda good to see it so that it is discoverable that you can do it and explain that you need to talk to your admin to enable it (as a tooltip. 

This is especially important, I think, for existing users. If suddenly the button disappears, nobody (especially users who could use the "Test connection" before) will read the release notes or documentation, we will be opening for floodgate of issues "The `test connection` button disappeared - please fix it".

Reply:
My suggestion will also be to add. "Talk to your deployment manager to enable it." or similar. I think it is time to start introducing the different kinds of users for our security model and once we explain the different kinds of users, use it consistently as messaging. 

I think having clear responsibilities "who is responsible for what" is also part of the user "education" we have to do a bit.  The average user will not know that they need to talk to the "deplomen manager" I think (if they are on a corporate installation or managed service). Which config? How to enable it? Where to find it ? 

I think there are often just "users" who who do not know (and should not even need to know actually) what Airflow configuration is. But if we tell them "talk to the guys that manage your Airflow", they will know what to do  (rather than trying to look on their own or create issues in airflow repo "I see this messsage, what should i do" (happens).

Reply:
Good point, but playing devils advocate :) I wonder if we need a non-boolean value for this then? What about the admin who chooses not to allow their users to test connections but constantly gets asked about it? Allowing admins to hide it could save them the same heartache.

Reply:
okay, so maybe we can give another config param to allow removing the button altogether? 
`remove_test_connection_button` set by default to `False` meaning this button will appear by default.

And `enable_test_connection=False` would control enabling/disabling the functionality

For evaluation:
remove_test_connection_button = Priority 0
enable_test_connection = Priority 1

0 has higher priority than 1. Would that make sense?

Reply:
or `test_connection_button` = enable/disable/remove. disable being the default?

Reply:
I'd strongly prefer a single config for this (we have enough config options already).

Reply:
Yeah. Very much agree with @jedcunningham  :). And agree that being able to disable it completely is a good idea from Deployment manager's perspective. I am torn between the defaults there though. Should new airflow has "remove" by defult or "disable". If only we could see if we are upgrading from older version or installing airflow from the scratch, that would be cool to have `remove` by default and 'disable` when you migrate.

Reply:
👍 🤔 👎 

I am ok with either approach as long as we have it off by default :)

Reply:
A bit confused about what could be the best approach here. But observed that the UI treats the values as a string anyway to do a comparison of the value. 

So, based on the conversation above and considering @jedcunningham's and @potiuk's suggestions, I thought to provide a single config param `test_connection` in the `core` section of Airflow configuration as this param needs to be used across Airflow UI, API and CLI. The supported values I have implemented are `Disabled`, `Enabled` and `Hidden` with `Disabled` being the default. @o-nikolas hope you are okay with this approach 🤔

Reply:
Yupp, that will be perfectly fine :)